src/com/serotonin/mango/rt/RuntimeManager.java

```diff
@@ -1,883 +1,889 @@
+import com.serotonin.mango.rt.dataSource.DataSourceRT;
+import com.serotonin.mango.rt.dataSource.meta.MetaDataSourceRT;
+import com.serotonin.mango.rt.event.SimpleEventDetector;
+import com.serotonin.mango.rt.event.compound.CompoundEventDetectorRT;
+import com.serotonin.mango.rt.event.detectors.PointEventDetectorRT;
+import com.serotonin.mango.rt.event.maintenance.MaintenanceEventRT;
+import com.serotonin.mango.rt.event.schedule.ScheduledEventRT;
+import com.serotonin.mango.rt.link.PointLinkRT;
+import com.serotonin.mango.rt.publish.PublisherRT;
+import com.serotonin.mango.util.DateUtils;
+import com.serotonin.mango.vo.DataPointVO;
+import com.serotonin.mango.vo.dataSource.DataSourceVO;
+import com.serotonin.mango.vo.event.CompoundEventDetectorVO;
+import com.serotonin.mango.vo.event.MaintenanceEventVO;
+import com.serotonin.mango.vo.event.PointEventDetectorVO;
+import com.serotonin.mango.vo.event.ScheduledEventVO;
+import com.serotonin.mango.vo.link.PointLinkVO;
+import com.serotonin.mango.vo.publish.PublishedPointVO;
+import com.serotonin.mango.vo.publish.PublisherVO;
+import com.serotonin.util.LifecycleException;
+import com.serotonin.web.i18n.LocalizableException;
+import com.serotonin.web.i18n.LocalizableMessage;
+
+public class RuntimeManager {
+	private static final Log LOG = LogFactory.getLog(RuntimeManager.class);
+
+	private final List<DataSourceRT> runningDataSources = new CopyOnWriteArrayList<DataSourceRT>();
+
+	/**
+	 * Provides a quick lookup map of the running data points.
+	 */
+	private final Map<Integer, DataPointRT> dataPoints = new ConcurrentHashMap<Integer, DataPointRT>();
+
+	/**
+	 * The list of point listeners, kept here such that listeners can be
+	 * notified of point initializations (i.e. a listener can register itself
+	 * before the point is enabled).
+	 */
+	private final Map<Integer, DataPointListener> dataPointListeners = new ConcurrentHashMap<Integer, DataPointListener>();
+
+	/**
+	 * Store of enabled event detectors.
+	 */
+	private final Map<String, SimpleEventDetector> simpleEventDetectors = new ConcurrentHashMap<String, SimpleEventDetector>();
+
+	/**
+	 * Store of enabled compound event detectors.
+	 */
+	private final Map<Integer, CompoundEventDetectorRT> compoundEventDetectors = new ConcurrentHashMap<Integer, CompoundEventDetectorRT>();
+
+	/**
+	 * Store of enabled publishers
+	 */
+	private final List<PublisherRT<?>> runningPublishers = new CopyOnWriteArrayList<PublisherRT<?>>();
+
+	/**
+	 * Store of enabled point links
+	 */
+	private final List<PointLinkRT> pointLinks = new CopyOnWriteArrayList<PointLinkRT>();
+
+	/**
+	 * Store of maintenance events
+	 */
+	private final List<MaintenanceEventRT> maintenanceEvents = new CopyOnWriteArrayList<MaintenanceEventRT>();
+
+	private boolean started = false;
+
+	//
+	// Lifecycle
+	synchronized public void initialize(boolean safe) {
+		if (started)
+			throw new ShouldNeverHappenException(
+					"RuntimeManager already started");
+
```

Comment:
Please not to change the formatting for this file

Reply:
I did not change any of formatting.That is not my fault, if github shows ( as I saw) many of strange changes, which I did not done.

Reply:
Difference in files

Your file has the end of the LF linux line (such a target) and the previous file had the end of the windows line. 

![Screenshot from 2019-11-06 15-32-00](https://user-images.githubusercontent.com/1280575/68307446-8b052800-00ab-11ea-82eb-72eeb5e209a7.png)

Reply:
I do not know why, because I downloaded intelij software for 64 bit linux system and it works under VM (Ubuntu) so as we talk yesterday, that file propably did not have any change since somebody under Windows worked in this file.